src/mo_frontend/menhir_error_reporting.ml

```diff
@@ -154,21 +154,22 @@ module Make
      checkpoint. If a syntax error is detected, we go back to this checkpoint
      and analyze it in order to produce a meaningful diagnostic. *)
 
-  exception Error of (Lexing.position * Lexing.position) * explanation list
-
-  let entry (start : 'a I.checkpoint) lexer =
-    let fail (inputneeded : 'a I.checkpoint) (checkpoint : 'a I.checkpoint) =
-      (* The parser signals a syntax error. Note the position of the
-         problematic token, which is useful. Then, go back to the
-         last [InputNeeded] checkpoint and investigate. *)
-      match checkpoint with
-      | HandlingError env ->
-          let (startp, _) as positions = positions env in
-          raise (Error (positions, investigate startp inputneeded))
-      | _ ->
-          assert false
-    in
-    I.loop_handle_undo Fun.id fail lexer start
+  (* exception Error of (Lexing.position * Lexing.position) * explanation list *)
+
+  (* TODO: remove to give control of parsing outside *)
+  (* let entry (start : 'a I.checkpoint) lexer = *)
+  (*   let fail (inputneeded : 'a I.checkpoint) (checkpoint : 'a I.checkpoint) = *)
+  (*     (\* The parser signals a syntax error. Note the position of the *)
+  (*        problematic token, which is useful. Then, go back to the *)
+  (*        last [InputNeeded] checkpoint and investigate. *\) *)
+  (*     match checkpoint with *)
+  (*     | HandlingError env -> *)
+  (*         let (startp, _) as positions = positions env in *)
+  (*         raise (Error (positions, investigate startp inputneeded)) *)
+  (*     | _ -> *)
+  (*         assert false *)
+  (*   in *)
+  (*   I.loop_handle_undo Fun.id fail lexer start *)
```

Comment:
What is meant by "parsing outside"? If we'll leave some commented code, we should be clear why it's there and we'd possibly want it back, or remove it if we won't need it.

Anyway, I see this was moved to `parsing.ml`, so perhaps we don't need it.

Reply:
"outside" is `parsing.ml` I should have it removed.